service/kernel/oneapi/cl_kernels/select_indexed.cl

```diff
@@ -29,102 +29,213 @@
 DECLARE_SOURCE(
     quick_select_simd,
 
-    algorithmFPType get_rnd(__global const algorithmFPType * rndseq, int RndPeriod, int * count) {
-        algorithmFPType ret = rndseq[(*count)++];
-        if (*count >= RndPeriod) *count = 0;
+    algorithmFPType get_rnd(__global const algorithmFPType * rnd_seq, int RndPeriod, int * count) {
+        algorithmFPType ret = rnd_seq[(*count)++];
+        if (*count >= RndPeriod)
+        {
+            *count = 0;
+        }
         return ret;
     }
 
-    void partition(__global algorithmFPType * pdist, __global int * plabels, int start, int end, int local_id, int local_size, algorithmFPType pivot,
-                   int * split, int * total_big_num) {
-        const int global_id = get_global_id(0);
-        int full_size       = end - start;
+    void partition_by_values(__global algorithmFPType * values, __global int * indices, int partition_start, int partition_end, int local_id,
+                             int local_size, algorithmFPType pivot, int * split_index_ptr, int * great_total_ptr) {
+        int full_size       = partition_end - partition_start;
         int last_group_size = full_size % local_size;
         int full_group_size = full_size - last_group_size;
-        for (int i = start + local_id; i < end; i += local_size)
+
+        for (int i = partition_start + local_id; i < partition_end; i += local_size)
```

Comment:
is int enough for your purposes in cl files? may be it should be unsigned at least?

Reply:
Sigined int is enough and is recommended to be used due to better performance